src/sentry/integrations/github/repository.py

```diff
@@ -82,35 +68,74 @@ def compare_commits(self, repo, start_sha, end_sha):
 
         # use config name because that is kept in sync via webhooks
         name = repo.config['name']
-        if start_sha is None:
-            try:
+        try:
+            if start_sha is None:
                 res = client.get_last_commits(name, end_sha)
-            except Exception as e:
-                installation.raise_error(e)
+                return self._format_commits(client, name, res[:20])
```

Comment:
is there a reason we increase the last commits to 20? Not sure if it matters, just wondering

Reply:
I was trying to match the number of commits we get in webhooks.